### PR TITLE
feat(javadoc): remove arm64 selector to rebuild the nodepool with taint

### PR DESCRIPTION
**Pull request by smerleCB**
as per https://github.com/jenkins-infra/helpdesk/issues/3623

to be used with https://github.com/jenkins-infra/azure/pull/418 as a new node pool with a taint

### CODE REVIEW

**Review by dduportal — CHANGES_REQUESTED** [review]
Can't merge as it: this change might  migrate the workload on an intel workload which might be expected?

It does not follow the steps described in https://github.com/jenkins-infra/helpdesk/issues/3623 from my understanding: you first need a second arm64 node pool

**Comment by smerleCB** [comment]
> Can't merge as it: this change might migrate the workload on an intel workload which might be expected?
> 
> It does not follow the steps described in [jenkins-infra/helpdesk#3623](https://github.com/jenkins-infra/helpdesk/issues/3623) from my understanding: you first need a second arm64 node pool

this application (javadoc) can spawn on either intel or arm64, as it's the only one on the pool right now, removing the selector should allow it to spawn on the intel while we rebuild the arm64 nodepool or I am mistaking ?

**Comment by dduportal** [comment]
As seen together, the issue need to be slighlty updated to express the plan we had a consensus in the 1:1 meeting (Tl;DR; new node poole arm64 with taints, update PR here to relax constraint about the nodepool name andadd the tolerations immediatly, etc.)

File: config/javadoc.yaml [hunk 1]
```diff
@@ -32,6 +32,13 @@ htmlVolume:
 
 replicaCount: 2
 
+# hard selector to force usage of arm64 node pool to save money
```
Comment (dduportal) [inline config/javadoc.yaml:35]:
```suggestion
# Specify the "hard" scheduling constraints
```

File: config/javadoc.yaml [hunk 2]
```diff
@@ -32,6 +32,13 @@ htmlVolume:
...
 nodeSelector:
-  kubernetes.azure.com/agentpool: arm64small
   kubernetes.io/arch: arm64
```
Comment (dduportal) [inline config/javadoc.yaml:37]:
```suggestion
  # Ensure ARM64 is used to decrease cloud bill (instead of using `amd64`
  kubernetes.io/arch: arm64
```

**Comment by dduportal** [comment]
Setting it to draft (ref. https://github.com/jenkins-infra/helpdesk/issues/3623#issuecomment-1613504371) until @smerleCB @smerle33 is back from vacations

**Review by dduportal — APPROVED** [review]

**Comment by dduportal** [comment]
- Code is 👍 and the toleration has been tested in #4206 
- Helmfile diff is 👍

**Pull request merged**